crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -465,7 +466,14 @@ impl CumulocityConverter {
                 messages.push(message);
             } else {
                 // The message must be sent over HTTP
-                let _ = self.http_proxy.send_event(c8y_event).await?;
+                let create_event = CreateEvent {
+                    event_type: c8y_event.event_type,
```

Comment:
It looks a bit weird that we first create a `C8yCreateEvent` and then create this `CreateEvent` out of it. I understand the need to create a different representations with `externalSource` for JSON over MQTT and `source` for HTTP. We should only be needing just one struct where that source field can be an enum of either `externalSource` or `source`. If this isn't trivial, you may address that refactoring in a different PR.

Reply:
I will not address the suggestion in this PR, but created an issue https://github.com/thin-edge/thin-edge.io/issues/2429